File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 12/7/2025, 1:14:38 AM
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        nm=[]
        nm2=[]
        nm3=[]
        l=len(nums)
        for i in range(l):
            if i==0:
                nm.append(1)
            else:
                nm.append(nm[i-1]*nums[i-1])
        for i in range(l-1,-1,-1):
            if i==l-1:
                nm2.append(1)
            else:
                nm2.append(nm2[-1]*nums[i+1])
        nm2=nm2[::-1]
        for i in range(l):
            nm3.append(nm[i]*nm2[i])
        return nm3

File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.cpp
// Last updated: 4/4/2026, 12:41:56 AM
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n;
    n=nums.size();
    vector <int> mlt;
    int lft=1;
    for (int i=0;i<n;i++){
        mlt.push_back(lft);
        lft*=nums[i];
        
    }
    int right=1;
    for (int i=n-1;i>=0;i--){
        mlt[i]*=right;
        right*=nums[i];
    }
    return mlt;
    }
};